mxcube3/routes/qutils.py

```diff
@@ -140,14 +140,140 @@ def queue_to_json_response(node=None):
+    failed = entry._execution_failed
+
+    curr_entry = mxcube.queue.queue_hwobj.get_current_entry()
+
+    if failed:
+        state = 3
+    elif executed:
+        state = 2
+    elif mxcube.queue.queue_hwobj.is_executing and (curr_entry == entry or curr_entry == entry._parent_container):
+        state = 1
+    else:
+        state = 0
+
+    return (enabled, state)
+
+
+def get_queue_state():
+    """
+    Return the dictionary representation of the queue state plus some extra lists.
+
+    :returns: dictionary on the form:
+              {
+                "history": [],
+                "loaded": "1:01",
+                "sample_list": [ sampleID_1, sampleID_2 ...],
+                "todo": [ sampleID_1, sampleID_2, ...],
+                "queue": {sampleID_1: [task1, ... taskn]
+                        .
+                        .
+                        .
+                        sampleID_n: [task1, ... taskn] }
+                "sample_list": {......}
+               }
+    """
+    if mxcube.diffractometer.use_sc:
+        samples_list_sc = mxcube.sample_changer.getSampleList()
+        samples = {}
+
+        for s in samples_list_sc:
+            sample_dm = s.getID() or ""
+            sample_data = {"sampleID": s.getAddress(),
+                           "location": ":".join(map(str, s.getCoords())),
+                           "sampleName": "Sample-%s" % s.getAddress().replace(':', ''),
+                           "code": sample_dm,
+                           "type": "Sample"}
+
+            sample_data["defaultPrefix"] = limsutils.get_default_prefix(sample_data, False)
+            samples.update({s.getAddress(): sample_data})
+
+    queue = queue_to_dict()
+    try:
+        order = queue.pop("sample_order")
+    except Exception:
+        order = []
+        pass
+
+    sample_list_ho = mxcube.queue.queue_hwobj._queue_entry_list
+    todo = []
+    history = []
+    sample_list = {}
+    for sample in sample_list_ho:
+        sample_model = sample.get_data_model()
+        name = sample_model.loc_str
+        if queue[name]["state"] > 1:
+            history.append(name)
+        else:
+            todo.append(name)
+
+        sample_data = {"sampleID": name,
+                       "location": "Manual" if sample_model.free_pin_mode else sample_model.loc_str,
+                       "sampleName": sample_model.get_name(),
+                       "type": "Sample",
+                       "proteinAcronym": sample_model.crystals[0].protein_acronym,
+                       "defaultPrefix": sample_model.get_name() + '-' + sample_model.crystals[0].protein_acronym
+                       }
+
+        sample_list.update({name: sample_data})
+
+    if mxcube.diffractometer.use_sc:
+        if len(sample_list) != 0:
+            sample_list = samples
+        else:
+            sample_list = {}
+        try:
+            basket, puck = mxcube.queue.mounted_sample.split(':')  #sample_changer.getLoadedSample().split(':')
```

Comment:
Sorry to come back on this... As already said in previous review, we can't split in 2 mounted sample location, since location can have any number of containers.